src/Base/PyHandle.h

```diff
@@ -0,0 +1,152 @@
+/***************************************************************************
+ *   Copyright (c) 2019 Viktor Titov (DeepSOIC) <vv.titov@gmail.com>       *
+ *                                                                         *
+ *   This file is part of the FreeCAD CAx development system.              *
+ *                                                                         *
+ *   This library is free software; you can redistribute it and/or         *
+ *   modify it under the terms of the GNU Library General Public           *
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef FREECAD_BASE_PYHANDLE_H
+#define FREECAD_BASE_PYHANDLE_H
```

Comment:
My understanding is that modern C++ prefers to use `#pragma once` versus these more familiar header tags.

Reply:
the rest of freecad uses macro guards, so I didn't change it. That being said, there is a particular bug in Qt Creator that falsely complains about header guards in circular includes. Which made me put #pragma once in a few places to make it show me all the real-time compilation goodness.

Reply:
> My understanding is that modern C++ prefers to use #pragma once versus these more familiar header tags.

No, that's wrong. It's true that msvc, gcc, clang and many other C++ compilers support the pragma once construct but as far as I know it's not defined in the C++ standard. And I wonder whether gcc or clang complains about it when compiling with -Wpedantic because it's not strict ISO C++. So, because of this and the fact that we use file guards everywhere in FreeCAD I recommend to also use it here.

A game changer will be the move to C++20 (in several years) which introduces the new concept of modules. This will stop the separation of header and source files and adds a new keyword called "import". The main benefit is supposed to considerably reduce the compile time.

Reply:
Ah, well then I guess my understand was wrong 😜

I'm not exactly looking forward to C++20, but only because I'm old and
don't like change 😂